### PR TITLE
Add distinguish File name for XML file's accessionIds

### PR DESCRIPTION
**Pull request by hannyle**
### Description

Similar to object's forms, XML files should also have a distinct ` - File name` next to linked `accessionId` to separate them.

### Related issues

Additional solution for XML files in issue https://github.com/CSCfi/metadata-submitter-frontend/issues/436

### Type of change

- [x] New feature (non-breaking change which adds functionality)

### Changes Made

- Add `- File name` of XML files next to linked `accessionId`
- Create a custom hook for `getLinkedDereferencedSchema` and reorganise some functions in `WizardFillObjectDetailsForm` 

### Testing

- [x] Integration Tests

### CODE REVIEW

**Review by lilachic — APPROVED** [review]
LGTM.

File: src/components/NewDraftWizard/WizardForms/WizardLinkedDereferencedSchemaHook.js [hunk 1]
```diff
@@ -0,0 +1,174 @@
+//@flow
```
[the remark this answers is not in the document]
Comment (hannyle) [reply src/components/NewDraftWizard/WizardForms/WizardLinkedDereferencedSchemaHook.js]: Good notice. I forgot we had a `WizardHooks` folder. I moved the file.

**Review by blankdots — APPROVED** [review]

**Comment by blankdots** [comment]
seems it needs a rebase fore merging

**Comment by blankdots** [comment]
seems that replacing a file gives out an error in the front-end even though backend responds ok:
```
metadata_submitter_backend_dev | [2021-09-28 10:17:46][server][1 MainProcess][DEBUG   ](L:427) _format_data_to_replace_and_add_to_db: Operator formatted data for study to add to DB
metadata_submitter_backend_dev | [2021-09-28 10:17:46][server][1 MainProcess][DEBUG   ](L:91) exists: DB check exists for ca2c7b284ea64accb025fa23d499bb2b in collection study.
metadata_submitter_backend_dev | [2021-09-28 10:17:46][server][1 MainProcess][DEBUG   ](L:233) replace: DB doc in study replaced with {'alias': 'GSE10966', 'centerName': 'GEO', 'identifiers': {'primaryId': 'SRP000539', 'externalId': [{'label': 'primary', 'namespace': 'BioProject', 'value': 'PRJNA108793'}, {'namespace': 'GEO', 'value': 'GSE10966'}]}, 'descriptor': {'studyTitle': 'Different title for testing purposes', 'studyType': 'Other', 'studyAbstract': 'Part of a set of highly integrated epigenome maps for Arabidopsis thaliana. Keywords: Illumina high-throughput bisulfite sequencing Overall design: Whole genome shotgun bisulfite sequencing of wildtype Arabidopsis plants (Columbia-0), and met1, drm1 drm2 cmt3, and ros1 dml2 dml3 null mutants using the Illumina Genetic Analyzer.', 'centerProjectName': 'GSE10966'}, 'studyLinks': [{'xrefDb': 'pubmed', 'xrefId': '18423832'}], 'studyAttributes': [{'tag': 'parent_bioproject', 'value': 'PRJNA107265'}], 'accessionId': 'ca2c7b284ea64accb025fa23d499bb2b', 'dateModified': datetime.datetime(2021, 9, 28, 10, 17, 46, 545233), 'dateCreated': datetime.datetime(2021, 9, 28, 10, 17, 44, 513000), 'publishDate': datetime.datetime(2021, 11, 28, 10, 17, 44, 513000)} for ca2c7b284ea64accb025fa23d499bb2b.
metadata_submitter_backend_dev | [2021-09-28 10:17:46][server][1 MainProcess][DEBUG   ](L:554) _format_data_to_replace_and_add_to_db: XMLOperator formatted data for xml-study to add to DB
metadata_submitter_backend_dev | [2021-09-28 10:17:46][server][1 MainProcess][DEBUG   ](L:91) exists: DB check exists for ca2c7b284ea64accb025fa23d499bb2b in collection xml-study.
metadata_submitter_backend_dev | [2021-09-28 10:17:46][server][1 MainProcess][DEBUG   ](L:233) replace: DB doc in xml-study replaced with {'accessionId': 'ca2c7b284ea64accb025fa23d499bb2b', 'content': '<STUDY_SET>\n    <STUDY alias="GSE10966" center_name="GEO">\n        <IDENTIFIERS>\n            <PRIMARY_ID>SRP000539</PRIMARY_ID>\n            <EXTERNAL_ID label="primary" namespace="BioProject">PRJNA108793</EXTERNAL_ID>\n            <EXTERNAL_ID namespace="GEO">GSE10966</EXTERNAL_ID>\n        </IDENTIFIERS>\n        <DESCRIPTOR>\n            <STUDY_TITLE>Different title for testing purposes\n            </STUDY_TITLE>\n            <STUDY_TYPE existing_study_type="Other" />\n            <STUDY_ABSTRACT>Part of a set of highly integrated epigenome maps for Arabidopsis thaliana. Keywords:\n                Illumina high-throughput bisulfite sequencing Overall design: Whole genome shotgun bisulfite sequencing\n                of wildtype Arabidopsis plants (Columbia-0), and met1, drm1 drm2 cmt3, and ros1 dml2 dml3 null mutants\n                using the Illumina Genetic Analyzer.\n            </STUDY_ABSTRACT>\n            <CENTER_PROJECT_NAME>GSE10966</CENTER_PROJECT_NAME>\n        </DESCRIPTOR>\n        <STUDY_LINKS>\n            <STUDY_LINK>\n                <XREF_LINK>\n                    <DB>pubmed</DB>\n                    <ID>18423832</ID>\n                </XREF_LINK>\n            </STUDY_LINK>\n        </STUDY_LINKS>\n        <STUDY_ATTRIBUTES>\n            <STUDY_ATTRIBUTE>\n                <TAG>parent_bioproject</TAG>\n                <VALUE>PRJNA107265</VALUE>\n            </STUDY_ATTRIBUTE>\n        </STUDY_ATTRIBUTES>\n    </STUDY>\n</STUDY_SET>'} for ca2c7b284ea64accb025fa23d499bb2b.
metadata_submitter_backend_dev | [2021-09-28 10:17:46][server][1 MainProcess][INFO    ](L:68) replace_metadata_object: Replacing object with schema study to database succeeded with accession id: ca2c7b284ea64accb025fa23d499bb2b
metadata_submitter_backend_dev | [2021-09-28 10:17:46][server][1 MainProcess][INFO    ](L:431) put_object: PUT object with accession ID ca2c7b284ea64accb025fa23d499bb2b in schema study was successful.
metadata_submitter_backend_dev | [2021-09-28 10:17:46][server][1 MainProcess][INFO    ](L:199) _check_csrf: Skipping Referer check due to request coming from frontend.
metadata_submitter_backend_dev | [2021-09-28 10:17:46][server][1 MainProcess][DEBUG   ](L:91) exists: DB check exists for cb296bb825584ecd880c5ab17ac9f48e in collection folder.
metadata_submitter_backend_dev | [2021-09-28 10:17:46][server][1 MainProcess][INFO    ](L:497) _check_patch_folder: replace on tags in folder
metadata_submitter_backend_dev | [2021-09-28 10:17:46][server][1 MainProcess][DEBUG   ](L:260) query: DB doc query performed in user.
metadata_submitter_backend_dev | [2021-09-28 10:17:46][server][1 MainProcess][INFO    ](L:874) check_user_has_doc: found doc cb296bb825584ecd880c5ab17ac9f48e at current user
metadata_submitter_backend_dev | [2021-09-28 10:17:46][server][1 MainProcess][DEBUG   ](L:147) patch: DB doc in folder patched for cb296bb825584ecd880c5ab17ac9f48e with data [{'op': 'replace', 'path': '/drafts/1/tags', 'value': {'submissionType': 'XML', 'fileName': 'No file name', 'displayTitle': 'No file name'}}].
metadata_submitter_backend_dev | [2021-09-28 10:17:46][server][1 MainProcess][INFO    ](L:748) update_folder: Updating folder with id cb296bb825584ecd880c5ab17ac9f48e to database succeeded.
metadata_submitter_backend_dev | [2021-09-28 10:17:46][server][1 MainProcess][INFO    ](L:658) patch_folder: PATCH folder with ID cb296bb825584ecd880c5ab17ac9f48e was successful.

```

while frontend shows:
thumb-Clipboard - 2021 Sep 28, 13_24

--- 


another issue, replacing a file with the same name fails:
Peek 2021-09-28 13-30

**Comment by hannyle** [comment]
@blankdots the first issue was merging error, I fixed that.
For the second error I would create a new issue for it and fix in another PR if that's fine.

**Comment by blankdots** [comment]
a new issue sounds very good!

**Pull request merged**